### PR TITLE
Allow libraries to define their own locale files

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
Addresses: https://github.com/blackbaud/skyux2/issues/1546

NOTE: I had to fix some unit tests that were reaching a bit too far into other files.

### CODE REVIEW

File: config/sky-pages/sky-pages.config.js [hunk 1]
```diff
@@ -22,132 +22,137 @@ function readConfig(file) {
   return fs.readJsonSync(file, 'utf8');
 }
 
-module.exports = {
+/**
```
Comment (Blackbaud-SteveBrush) [inline config/sky-pages/sky-pages.config.js]: There are zero functional changes to this file; I simply adjusted the format so we can use a few of its methods (namely `spaPath`) via destructuring. (See my changes to `common.webpack.config.js` for an example.)

File: test/sky-pages-module-generator.spec.js [hunk 2]
```diff
@@ -1,24 +1,73 @@
 /*jshint jasmine: true, node: true */
 'use strict';
 
-const codegen = require('../utils/codegen-utils');
+const mock = require('mock-require');
```
Comment (Blackbaud-SteveBrush) [inline test/sky-pages-module-generator.spec.js]: This spec file was affecting the state of other unit tests, so I mocked out its dependencies and made sure those dependencies' unit tests were covered 100%.

**Comment by Blackbaud-SteveBrush** [comment]
@Blackbaud-BobbyEarl This is ready for another look.

File: config/sky-pages/sky-pages.config.js (hunk 1 above)
Comment (Blackbaud-BobbyEarl) [reply config/sky-pages/sky-pages.config.js]: Just want to say I appreciate you calling this out.  It definitely helps when reviewing.

**Comment by Blackbaud-BobbyEarl** [comment]
Great job on this @Blackbaud-SteveBrush!

Aside from the destructing we talked about offline the only other thing that stood out to me is the use of the `try/catch`.  That masks whether there was an error because the file doesn't exist, or because of another error.  I'd prefer to instead use the existsSync and let any other fatal error bubble up.

**Comment by Blackbaud-SteveBrush** [comment]
@Blackbaud-BobbyEarl I've since removed the destructured method imports, and removed the `try/catch` blocks for the `fs` operations. Let me know if I missed anything!

**Pull request merged**